src/StreamJsonRpc/Reflection/MessageFormatterRpcMarshaledContextTracker.cs

```diff
@@ -87,6 +90,62 @@ private enum MarshalMode
             MarshallingRealObject = 1,
         }
 
+        internal static bool TryGetMarshalOptionsForType(Type type, [NotNullWhen(true)] out JsonRpcProxyOptions? proxyOptions, [NotNullWhen(true)] out JsonRpcTargetOptions? targetOptions)
+        {
+            proxyOptions = null;
+            targetOptions = null;
+            if (type.IsInterface is false)
+            {
+                return false;
+            }
+
+            if (MarshaledTypes.TryGetValue(type, out (JsonRpcProxyOptions ProxyOptions, JsonRpcTargetOptions TargetOptions) options))
+            {
+                proxyOptions = options.ProxyOptions;
+                targetOptions = options.TargetOptions;
+                return true;
+            }
+
+            foreach ((Type implicitlyMarshaledType, JsonRpcProxyOptions typeProxyOptions, JsonRpcTargetOptions typeTargetOptions) in ImplicitlyMarshaledTypes)
+            {
+                if (implicitlyMarshaledType == type ||
+                    (implicitlyMarshaledType.IsGenericTypeDefinition &&
+                     type.IsConstructedGenericType &&
+                     implicitlyMarshaledType == type.GetGenericTypeDefinition()))
+                {
+                    proxyOptions = typeProxyOptions;
+                    targetOptions = typeTargetOptions;
+                    MarshaledTypes.TryAdd(type, (proxyOptions, targetOptions));
+                    return true;
+                }
+            }
+
+            if (type.GetCustomAttribute<RpcMarshalableAttribute>() is not null)
+            {
+                if (typeof(IDisposable).IsAssignableFrom(type) is false)
```

Comment:
Do we need to check for default interface methods, can we even check for it? I was thinking it could be trouble if client/server has slightly different versions where the default implementation is different. Although it wouldn't be a problem specific to marshaled interface case.

Reply:
Discussed in person. Any consideration regarding default methods in interfaces should be addressed across both marshaling interfaces and "normal" server/client proxies as they should behave in the same way.